data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -61,15 +72,35 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
                 throw new PatternSyntaxException("field_delimiter_regex is not a valid regex string", keyValueProcessorConfig.getFieldDelimiterRegex(), -1);
             }
 
+            if (keyValueProcessorConfig.getRecursive()) {
+                if (Pattern.compile(keyValueProcessorConfig.getFieldDelimiterRegex()).matcher(delimiterBracketCheck).matches()) {
+                    throw new IllegalArgumentException("The set field delimiter regex cannot contain brackets while you are trying to recurse.");
+                }
+                if (keyValueProcessorConfig.getFieldDelimiterRegex().length() != 1) {
```

Comment:
The length of regex expression should not be limited to 1, e.g. the following regex expression: 

```
[&|]
```
could match either `&` or `|` as delimiter even if the length of the delimiter is 1